xoinvader/background.py

```diff
@@ -32,24 +29,22 @@ def name(self):
     def lines(self):
         return self._lines
 
-    @property
-    def length(self):
-        return self._length
-
     def add_line(self, line):
-        """Add line to `lines` list of the chunk. Increases `length` by 1.
+        """Add line to `lines` list of the chunk.
 
         :param str line: line to add
         """
         self._lines.append(line)
-        self._length += 1
 
     def __getitem__(self, index):
         return self._lines[index]
 
     def __setitem__(self, index, item):
         self._lines[index] = item
 
+    def __len__(self):
+        return len(self._lines)
```

Comment:
Why do you think that len take O(n)?
len calls PySIZE macro.
https://github.com/python/cpython/blob/9530f485920ebb1404d4adaf24e6f3f7ddbe9622/Objects/listobject.c#L397
It defines here: https://github.com/python/cpython/blob/d6a5199175066149c73f62ee020a3161e504e1cf/Include/object.h#L119
And